tests/test_all

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
+
+# Runs pre-commit, then all the scripts in this dir, and then builds documentaiton locally
+
+# Get the list of script files in the directory
+script_files=("$(dirname "$(realpath "$0")")"/*)
+num_files=$((${#script_files[@]} + 1))
+error_scripts=()
+
+echo "Found $num_files tests"
+
+echo "----- Running test 1 of $num_files: pre-commit"
+pre-commit run --all-files
+
+# Iterate over script files in the directory
+for script_file in "${script_files[@]}"; do
+
+  # Exclude the current script and non-executable files from execution
+  if [ "$(basename "$script_file")" != "test_all" ]; then
+    if [ -x "$script_file" ]; then
+      # Update progress
+      echo "----- Running test $((i + 1)) of $num_files: $script_file"
```

Comment:
`i` doesn't seem to be updated, so all of my "------ Running" lines start with `----- Running test 1 of 5` and I never see `2 of 5`.

Reply:
good catch - i changed how the looping worked and forgot to update this part...fixing now